controllers/admin/AdminImportController.php

```diff
@@ -1659,49 +2251,61 @@ protected function productImportOne($info, $default_language_id, $id_lang, $forc
-                    ($lang_field_error = $manufacturer->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true &&
-                    !$validateOnly && // Do not move this condition: previous tests should be played always, but next ->add() test should not be played in validateOnly mode
-                    $manufacturer->add()) {
-                    $product->id_manufacturer = (int)$manufacturer->id;
-                    $manufacturer->associateTo($product->id_shop_list);
+        if (!empty($product->manufacturer)) {
+            if (Manufacturer::manufacturerExists((int)$product->manufacturer)) {
+                $product->id_manufacturer = (int)$product->manufacturer;
+            } elseif (is_string($product->manufacturer)) {
+                if ($manufacturer = Manufacturer::getIdByName($product->manufacturer)) {
+                    // Found manufacturer by name
+                    $product->id_manufacturer = (int)$manufacturer;
                 } else {
-                    if (!$validateOnly) {
-                        $this->errors[] = sprintf(
-                            $this->trans('%1$s (ID: %2$s) cannot be saved', array(), 'Admin.Advparameters.Notification'),
-                            $manufacturer->name,
-                            (isset($manufacturer->id) && !empty($manufacturer->id))? $manufacturer->id : 'null'
-                        );
-                    }
-                    if ($field_error !== true || isset($lang_field_error) && $lang_field_error !== true) {
-                        $this->errors[] = ($field_error !== true ? $field_error : '').(isset($lang_field_error) && $lang_field_error !== true ? $lang_field_error : '').
-                            Db::getInstance()->getMsgError();
+                    // Manufacturer creation
+                    $manufacturer         = new Manufacturer();
+                    $manufacturer->name   = $product->manufacturer;
+                    $manufacturer->active = true;
+                    // FIXME needing such a comment about "&& !$validateOnly" position is a code smell. Refacto needed.
+                    if (($fieldError = $manufacturer->validateFields(UNFRIENDLY_ERROR, true)) === true
+                        && ($langFieldError = $manufacturer->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true
+                        && !$validateOnly // Do not move this condition: previous tests should be played always, but next ->add() test should not be played in validateOnly mode
+                        && $manufacturer->add()
+                    ) {
+                        $product->id_manufacturer = (int)$manufacturer->id;
+                        $manufacturer->associateTo($product->id_shop_list);
+                    } else {
+                        if (!$validateOnly) {
+                            $this->errors[] = $this->trans('%1$s (ID: %2$s) cannot be saved', [
```

Comment:
Same comment about translation placeholder. Please use a variable name surrounded by `%%` an add it in the second parameter.

Reply:
Use long array syntax to create your arrays please.